python/ray/data/read_api.py

```diff
@@ -338,6 +340,86 @@ def read_datasource(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_mongo(
+    uri: str,
+    database: str,
+    collection: str,
+    *,
+    pipeline: Optional[List[Dict]] = None,
+    schema: Optional["pymongoarrow.api.Schema"] = None,
+    parallelism: int = -1,
+    ray_remote_args: Dict[str, Any] = None,
+    **mongo_args,
```

Comment:
How do we have name conflicts in general, what if mongo_args also has a parallelism arg? Should we accept `mongo_args: Dict[str, Any]`

Reply:
pymongoarrow will fail it since it doesn' have this arg.

Reply:
We can unpack it when we pass to pymongoarrow.